mediaproviders/src/main/java/com/kaltura/playkit/plugins/ott/PhoenixAnalyticsPlugin.java

```diff
@@ -163,7 +163,9 @@ public void addListeners() {
         messageBus.addListener(this, PlayerEvent.ended, event -> {
             printReceivedEvent(event);
             resetTimer();
-            sendAnalyticsEvent(PhoenixActionType.FINISH);
+            if (!isMediaFinished) {
```

Comment:
@GouravSna is there a chance this condition will break this event from being sent?

Reply:
There could be 2 scenarios where Finish event will be fired.
1. Media duration % timer == 0 (or < threshold), in this case our timer will hit and send the event. Then after it when PlayerEvent.ended will come, event will not be fired.
2. Media duration % timer > threshold then event will be fired froom PlayerEvent.ended.